CMakeLists.txt

```diff
@@ -709,6 +718,34 @@ function(AddApplication)
     set_property(TARGET ${A_Target} APPEND PROPERTY COMPILE_DEFINITIONS ${A_Definitions})
     set_target_properties(${A_Target} PROPERTIES OUTPUT_NAME "${A_ExecutableName}" PREFIX "" FOLDER "engine")
 
+    # Append Windows specific manifests.
+    if (WIN32)
+      set(WINDOWS_MANIFESTS_DIR ${ENGINE_DIR}/sys/manifests)
+      target_sources(${A_Target} PRIVATE
+          # App is DPI aware, so windows doesn't try to scale UI.
+          ${WINDOWS_MANIFESTS_DIR}/dpi-aware.manifest
+          # App supports Windows Vista+, no quirks required.
+          ${WINDOWS_MANIFESTS_DIR}/supported-os.manifest
+          # Use visual styles for Windows UI controls (message boxes, buttons, etc.).
+          ${CMAKE_BINARY_DIR}/${A_Target}/enable-visual-styles.manifest
+      )
+
+      # Setup variables for visual styles manifest.
+      set(DAEMON_APP_VERSION "0.0.51.1")
```

Comment:
Not sure about this.

Looks like some automation mechanism of getting version from release required?

Reply:
What is `DAEMON_APP_VERSION` used for, or to say it otherwise, what is expected to read this?

For your information, the places were we currently hardcode the version can be seen there:

https://github.com/DaemonEngine/Daemon/pull/458/files
https://github.com/Unvanquished/Unvanquished/pull/1392/files

Reply:
Required in visual styles manifest as application version.
If not set, Windows forbids to start the app.

https://github.com/DaemonEngine/Daemon/pull/457/files#diff-bf6463665cb899f7da56306be64ebc85aea4165bfbda0c2f512c7feb20984db9R5

Reply:
Is there some Microsoft/Industry usage that make you think the version would be `0.0.51.1` instead of `0.51.1.0`, If `0.0.0.0` is for `SOMETHING.MAJOR.MINOR.PATCH`, what is  `SOMETHING`?

Reply:
Major.Minor.Revision.Build

Reply:
https://semver.org/
Additional labels for pre-release and build metadata are available as extensions to the MAJOR.MINOR.PATCH format.

Microsoft decided to go 4 component version schema decades ago, so we have to follow in manifest styles.

Reply:
If major is 51 and minor is 1, should I use 51.1.0.0 than?

Reply:
Moved to https://github.com/DaemonEngine/Daemon/pull/462

Reply:
Currently we use that:

```
0.51.0
↑ ↑  ↑
⏐ ⏐  patch
⏐ ⏐
⏐ minor
⏐
major
```

Reply:
Done, thanks!